### ISSUE TITLE
BUG: offsets normalize and __eq__

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#17689]
Two otherwise-identical  `DateOffset`objects with different `normalize` attributes evaluate to equal.  But they do not behave equivalently:

```
In [1]: from pandas.tseries.offsets import *
In [2]: import pandas as pd
In [3]: ts = pd.Timestamp.now()
In [4]: off1 = DateOffset(days=1)
In [5]: off2 = DateOffset(days=1, normalize=True)

In [6]: off1 == off2
Out[6]: True

In [7]: ts + off1
Out[7]: Timestamp('2017-09-27 21:13:54.502976')

In [8]: ts + off2
Out[8]: Timestamp('2017-09-27 00:00:00')
```

These objects should not be considered equal.  If there's consensus, I'll make a PR.

**Comment by jreback** [comment]
this is probably a bug. we want to compare all params (though that is not entirely cheap).

**Comment by jbrockmendel** [comment]
Great.  I'll add it to the tslibs TODO list.  Don't worry about the cheapness; even without the cythonizing, there are some pretty significant speedups in the pipeline.

**Comment by jbrockmendel** [comment]
Does fixing the normalize comparison count as a breaking change that needs some special treatment, or can I just go ahead and fix it?

**Comment by jbrockmendel** [comment]
Closed by #21404
